Use "nested flow" in place of "subflow" in the docs (#14122)
PrefectHQ/prefect: docs/2.19.x/concepts/states.mdx

```

The same rule applies for a nested flow:

```python
@flow 

> def nested flow():

Reviewer: ```suggestion
def nested_flow():
```

    return 42 

@flow 
def my_flow():

>     result = nested flow() # return data

Reviewer: ```suggestion
    result = nested_flow() # return data
```

```

The same rule applies for a nested flow:

```python
@flow 

> def nested flow():

Reviewer: ```suggestion
def nested_flow():
```

    return 42 

@flow 
def my_flow():

>     state = nested flow(return_state=True) # return State

Reviewer: ```suggestion
    state = nested_flow(return_state=True) # return State
```

The final state of a flow is determined by its return value. The following rules apply:

*   If an exception is raised directly in the flow function, the flow run is marked as `FAILED`.

> *   If the flow does not return a value (or returns `None`), its state is determined by the states of all of the tasks and nested flows within it.

Reviewer: ```suggestion
*   If the flow does not return a value (or returns `None`), its state is determined by the states of all of the runs within it.
```

> *   If _any_ task run or nested flow run failed and none were cancelled, then the final flow run state is marked as `FAILED`.

Reviewer: ```suggestion
*   If _any_ run failed and none were cancelled, then the final flow run state is marked as `FAILED`.
```

> *   If _any_ task run or nested flow run was cancelled, then the final flow run state is marked as `CANCELLED`.

Reviewer: ```suggestion
*   If _any_ run was cancelled, then the final flow run state is marked as `CANCELLED`.
```
